data_collection/gazette/spiders/rj/rj_marica.py

```diff
@@ -0,0 +1,70 @@
+import re
+from datetime import date, datetime as dt
+
+import scrapy
+from dateutil.relativedelta import relativedelta
+from dateutil.rrule import MONTHLY, rrule
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class RjMaricaSpider(BaseGazetteSpider):
+    TERRITORY_ID = "3302700"
+    name = "rj_marica"
+    allowed_domains = ["marica.rj.gov.br"]
+    start_date = date(2006, 7, 17)
+    BASE_URL = "https://www.marica.rj.gov.br/wp-admin/admin-ajax.php?slug={GAZETTE_TYPE}&post_type={GAZETTE_TYPE}&year={YEAR}&month={MONTH}&action=alm_get_posts&order=DESC&orderby=date&posts_per_page=31"
+
+    def start_requests(self):
+        end_date = self.end_date + relativedelta(months=1)
+        for monthly_date in rrule(
+            freq=MONTHLY, dtstart=self.start_date, until=end_date
+        ):
```

Comment:
Aqui queria propor passar a implementar desse novo jeito...  

Entendi que fez a adição de um mês usando timedelta pq o rrule tem uma [*infeliz*] limitação de nem sempre incluir o end_date, o que, pra lógica mensal deste raspador, implica correr o risco de não incluir o mês vigente. Tem outros raspadores na base de código do repositório que tiveram que fazer contornos parecidos (a [base municipioonline](https://github.com/okfn-brasil/querido-diario/blob/d495f90193ae4699a9669cf735abc64f3c128eb6/data_collection/gazette/spiders/base/municipioonline.py#L35) é um caso). 

Como é um "problema comum", hoje parei para dar uma olhada na docs de rrule (um pouco surpresa de não ter uma opção pra isso) e achei essa outra possibilidade de implementação, usando [rruleset](https://dateutil.readthedocs.io/en/stable/rrule.html#dateutil.rrule.rruleset). Me pareceu melhorar a legibilidade.

O que vocês acham? @janainamendes29, @slfabio e, como tô propondo nesse caso particular mas visando adotar como padrão do repo a partir de agora, @ogecece tb. 
```suggestion
        monthly_dates = rruleset()
        monthly_dates.rrule(rrule(freq=MONTHLY, dtstart=self.start_date, until=self.end_date))
        monthly_dates.rdate(dt(self.end_date.year, self.end_date.month, self.end_date.day))
        
        for monthly_date in list(monthly_dates):
```

Reply:
Revendo aqui hoje, pensei que apenas 
```
      for monthly_date in rrule(
          freq=MONTHLY, 
          dtstart=self.start_date, 
          until=self.end_date + relativedelta(months=1)
      ):
```
poderia ser suficiente também, seria sucinto

Reply:
Galera, aqui a gente vai ter problemas em todas as soluções quando o `.start_date` for no final do mês nos dias 29, 30 e 31. Como nem todo mês tem esses dias, o `rrule()` vai pular esses meses com menos dias que o mês do `.start_date`. Por isso, é necessário puxar o `dtstart` pro primeiro dia do mês de `.start_date` pra garantir que todos os meses serão gerados.

Seguindo a mesma linha da primeira sugestão da @trevineju, já que o `rruleset` é interessante mesmo: 

```suggestion
        monthly_dates = rruleset()
        monthly_dates.rrule(rrule(MONTHLY, dtstart=self.start_date, until=self.end_date, bymonthday=[1]))
        monthly_dates.rdate(dt(self.start_date.year, self.start_date.month, 1))
        for monthly_date in monthly_dates:
```

O `bymonthday` faz com que todos as datas geradas sejam no primeiro dia do mês. Com isso, a gente garante que o mês final sempre entrará no conjunto (mesmo que o `.end_date` seja no dia 1 do mês) e não precisamos adicionar um mês a mais.

Mas aí geramos o problema do lado do `.start_date`. Se o `.start_date` for depois do primeiro dia do mês, o primeiro dia daquele mês não entrará no conjunto (ex: se `.start_date` for dia 01/03/2020, essa será a primeira data, mas se for 02/03/2020, a primeira data será apenas 01/04/2020). Assim, o `.rdate()` adicionando o primeiro dia do mês do `.start_date` resolve isso.

Isso tudo também poderia ser adaptado para a perspectiva do último dia do mês para que a lógica parecesse mais com o que a @janainamendes29 desenvolveu inicialmente, mas acho que a solução usando `.start_date` fica um pouco mais tranquila de implementar e entender.

E, por último, acho que seria bom a gente ter isso implementado no `utils` do projeto pra que qualquer pessoa possa usar sem precisar reimplementar sempre. Essa versão de iteração por mês e a versão de iteração por ano são bem comuns.

Reply:
deixei uma issue #1287